compiler/rustc_borrowck/messages.ftl

```diff
@@ -172,37 +500,59 @@ borrowck_suggest_create_freash_reborrow =
 borrowck_suggest_iterate_over_slice =
     consider iterating over a slice of the `{$ty}`'s content to avoid moving into the `for` loop
 
+borrowck_thread_local_outlive_function =
+    thread-local variable borrowed past end of function
+
 borrowck_ty_no_impl_copy =
     {$is_partial_move ->
         [true] partial move
         *[false] move
     } occurs because {$place} has type `{$ty}`, which does not implement the `Copy` trait
 
+borrowck_type_parameter_not_used_in_trait_type_alias =
+    type parameter `{$ty}` is part of concrete type but not used in parameter list for the `impl Trait` type alias
+
+borrowck_upvar_need_mut_due_to_borrow =
+    calling `{$place}` requires mutable binding due to mutable borrow of `{$upvar}`
+
+borrowck_upvar_need_mut_due_to_mutation =
+    calling `{$place}` requires mutable binding due to possible mutation of `{$upvar}`
+
 borrowck_use_due_to_use_closure =
     use occurs due to use in closure
 
 borrowck_use_due_to_use_generator =
     use occurs due to use in generator
 
+borrowck_used_here_by_closure =
+    used here by closure
+
 borrowck_used_impl_require_static =
     the used `impl` has a `'static` requirement
 
+borrowck_used_non_generic_for_generic =
+    used non-generic {$descr ->
+    [lifetime] lifetime
+    [type] type
+    *[constant] constant
```

Comment:
Ideally we'd split this into three messages rather than pattern matching on a string. We anticipate it would be difficult for translators to know what to match for in their translations. Likewise anywhere else you do this.